src/model/position.js

```diff
@@ -1057,3 +1066,12 @@ export default class Position {
  *
  * @typedef {String} module:engine/model/position~PositionStickiness
  */
+
+// Helper function used to inline text node access by using a cached parent.
+// Reduces the access to the Position#parent property 3 times (in total, when taken into account what #nodeAfter and #nodeBefore do).
+// See #6579.
+function getTextNode( position, positionParent ) {
+	const node = positionParent.getChild( positionParent.offsetToIndex( position.offset ) );
+
+	return ( node && node.is( 'text' ) && node.startOffset < position.offset ) ? node : null;
```

Comment:
I converted that to an `if()` statement. I didn't really see a good name for these conditions. Plus. The calculation of the offset should be done only if the first check is true, otherwise, we have a performance regression.